src/compiler/crystal/codegen/primitives.cr

```diff
@@ -274,6 +274,7 @@ class Crystal::CodeGenVisitor
     op_overflow = new_block "overflow"
     op_normal = new_block "normal"
 
+    builder.call(llvm_expect_i1_fun, [overflow_condition, llvm_false])
```

Comment:
You need to use the return value of the call in the branch instruction, no it's not documented, you need to read the code here: https://github.com/llvm-mirror/llvm/blob/master/lib/Transforms/Scalar/LowerExpectIntrinsic.cpp

Reply:
Alternatively we could emit [branch weight metadata](https://llvm.org/docs/BranchWeightMetadata.html) directly, but that's a pain and best left if/when LLVM actually removes the expect intrinsic/